feat(euclidean_cluster): use point_cloud_msg_wrapper

esteve opened the pull request:
## Description

<!-- Write a brief description of this PR. -->

This PR replaces the standard point cloud iterators from the `sensor_msgs` package with [Point Cloud Message Wrapper](https://gitlab.com/ApexAI/point_cloud_msg_wrapper)
## Related links

<!-- Write the links related to this PR. -->

https://github.com/autowarefoundation/autoware.universe/issues/1199

## Tests performed

<!-- Describe how you have tested this PR. -->

## Notes for reviewers

<!-- Write additional information if necessary. It should be written if there are related PRs that should be merged at the same time. -->

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

kenji-miyake commented:
@yukkysaito @aohsato @miursh Could you review this? :pray:

esteve commented:
@kenji-miyake sorry, this is still not ready, there are some more places in `euclidean_cluster` where I need to migrate to `point_cloud_msg_wrapper`

kenji-miyake commented:
Okay, when you are ready, please let them know again!

esteve commented:
@kenji-miyake @yukkysaito @aohsato @miursh this is now read for review, but I have a question about how to store RGB values (see https://github.com/autowarefoundation/autoware.universe/pull/1200#discussion_r915939898)

yukkysaito commented on perception/euclidean_cluster/lib/utils.cpp:
```diff
@@ -73,40 +75,22 @@ void convertObjectMsg2SensorMsg(
+    for (const auto & point : view) {
+      modifier.push_back(
+        {point.x, point.y, point.z,
+         static_cast<float>(
```
>each value is in a separate uint8_t

I think so too.
When PointXYZRGB is declared as follows, i think it's good PointXYZRGB make 4 uint8 for RGB. What do you think?
```
point_cloud_msg_wrapper::PointCloud2Modifier<autoware::common::types::PointXYZRGB> modifier
```
```
PointXYZRGB
- float32 X
- float32 Y
- float32X
- uint8 R
- uint8 G
- uint8 B
- uint8 NONE

```

mitsudome-r replied:
Do you have any intention why you add NONE at the end?

yukkysaito replied:
@mitsudome-r 
It is structure problem.
https://github.com/autowarefoundation/autoware.universe/pull/1200/files#diff-34905d424fde203854a98807e151655d568505124c1a5a8da63b428a4ed0a876L77-L79

I think PCL handle 4 bytes for fast access. so RGB field needs to use 3 bytes but it allocate 4 bytes.

esteve replied:
> I think PCL handle 4 bytes for fast access. so RGB field needs to use 3 bytes but it allocate 4 bytes.

That's true:

https://pointclouds.org/documentation/point__types_8hpp_source.html#l00287

PCL uses either a `float` or four `uint8_t` values to represent RGB values, I imagine that's why the point cloud in https://github.com/autowarefoundation/autoware.universe/pull/1200/files#diff-34905d424fde203854a98807e151655d568505124c1a5a8da63b428a4ed0a876L77-L79 uses a `FLOAT` to store RGB. I'll change it to four `uint8_t` as it seems it's compatible with PCL and at the same time more readable.

@yukkysaito thanks!

mitsudome-r replied:
@xmfcx Is finalizing documents for point types. We can come back to this once Fatih's task is over.

esteve commented:
@yukkysaito @mitsudome-r @kaancolak @xmfcx I've rebased this PR and fixed the conflicts, is it still waiting on  https://github.com/autowarefoundation/autoware.universe/pull/1200#issuecomment-1217813552 ?

mitsudome-r commented:
For the distortion correction, we [reverted](https://github.com/autowarefoundation/autoware.universe/pull/2317) migration to point_cloud_msg_wrapper.
Should we wait for [this](https://github.com/autowarefoundation/autoware-documentation/pull/180)  to be fixed?

miursh commented:
@mitsudome-r @esteve 
Do we still have the motivation to use this point_cloud_msg_wrapper? 
If not, would it be acceptable to close this PR for now?

miursh commented:
We are closing this issue for now since there has been no response for a month. Please feel free to reopen it if the issue still persists.

esteve commented:
@miursh thanks
